src/main/java/com/terraformersmc/terrestria/surface/CanyonSurfaceBuilder.java

```diff
@@ -0,0 +1,158 @@
+package com.terraformersmc.terrestria.surface;
+
+import com.mojang.datafixers.Dynamic;
+import com.terraformersmc.terraform.noise.OpenSimplexNoise;
+import net.minecraft.block.BlockState;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.world.biome.Biome;
+import net.minecraft.world.chunk.Chunk;
+import net.minecraft.world.gen.surfacebuilder.SurfaceBuilder;
+import net.minecraft.world.gen.surfacebuilder.TernarySurfaceConfig;
+
+import java.util.Random;
+import java.util.function.Function;
+
+public class CanyonSurfaceBuilder extends SurfaceBuilder<CanyonSurfaceConfig> {
+
+	private static final OpenSimplexNoise CLIFF_NOISE = new OpenSimplexNoise(346987);
```

Comment:
I am a tiny bit concerned about using the same canyon shape for every world, even if canyon biomes will likely end up at different positions.

Reply:
I can't see a way around this, I don't think it will be an issue